src/Sulu/Bundle/SecurityBundle/Command/SyncPhpcrPermissionsCommand.php

```diff
@@ -0,0 +1,147 @@
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\SecurityBundle\Command;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Sulu\Component\Content\Document\Behavior\SecurityBehavior;
+use Sulu\Component\Content\Document\Subscriber\SecuritySubscriber;
+use Sulu\Component\DocumentManager\Behavior\Mapping\UuidBehavior;
+use Sulu\Component\DocumentManager\Collection\QueryResultCollection;
+use Sulu\Component\DocumentManager\DocumentManagerInterface;
+use Sulu\Component\Security\Authentication\RoleInterface;
+use Sulu\Component\Security\Authorization\AccessControl\DoctrineAccessControlProvider;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+class SyncPhpcrPermissionsCommand extends Command
+{
+    protected static $defaultName = 'sulu:security:sync-phpcr-permissions';
+
+    /**
+     * @var EntityManagerInterface
+     */
+    private $entityManager;
+
+    /**
+     * @var DocumentManagerInterface
+     */
+    private $documentManager;
+
+    /**
+     * @var DoctrineAccessControlProvider
+     */
+    private $doctrineAccessControlProvider;
+
+    public function __construct(
+        EntityManagerInterface $entityManager,
+        DocumentManagerInterface $documentManager,
+        DoctrineAccessControlProvider $doctrineAccessControlProvider
+    ) {
+        $this->entityManager = $entityManager;
+        $this->documentManager = $documentManager;
+        $this->doctrineAccessControlProvider = $doctrineAccessControlProvider;
+
+        parent::__construct();
+    }
+
+    public function configure(): void
+    {
+        $this->setDescription('Sync object permissions from phpcr into database');
+        $this->setHelp(
+            'The <info>%command.name%</info> command syncs the object permissions of phpcr documents into the database.'
+        );
+    }
+
+    public function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        $roleIds = $this->getRoleIds();
+
+        if (0 === \count($roleIds)) {
+            $io->success('Nothing to do');
+
+            return 0;
+        }
+
+        $documents = $this->findDocuments($roleIds);
+        $amountDocuments = $documents->count();
+
+        if (0 === $amountDocuments) {
+            $io->success('Nothing to do');
+
+            return 0;
+        }
+
+        $io->progressStart($amountDocuments);
+
+        $count = 0;
+        foreach ($documents as $document) {
+            $io->progressAdvance();
+
+            if (!$document instanceof SecurityBehavior || !$document instanceof UuidBehavior) {
+                continue;
+            }
+
+            $this->doctrineAccessControlProvider->setPermissions(
+                SecurityBehavior::class,
```

Comment:
should we have here not a difference between pages and articles, else it could make problems when we migrating articles or pages away from phpcr as there is no real identifier which permission are pages and which are articles.

Reply:
I don't think we can do this easily now?

Reply:
But in the `PhpcrSecuritySubscriber`, we don't have the actual document. We just have `$type` which is `SecurityBehavior`

Reply:
Okay didn't know that. Then its okay for me to stay with `SecurityBehavior` at current state and check the permission migration when we are migrating pages to doctrine, will just make the update statement little bit more complex but at the end its just a join.